messybrainz/db/data.py

```diff
@@ -144,12 +144,14 @@ def submit_recording(connection, data):
     Returns:
         the Recording MessyBrainz ID of the data
     """
-    data_json = json.dumps(data, sort_keys=True, separators=(',', ':'))
-    data_sha256 = sha256(data_json.encode("utf-8")).hexdigest()
+    sha256_json = convert_to_messybrainz_json(data, transform=True)
```

Comment:
GAH! We're doing the serialization twice! Would it make sense to do it once and also kick out the lower case sha value as a side-effect that can be ignored?